src/components/Example1.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import '../styles/example1.css';
+import { addData, getData } from '../utils/indexedDB';
+
+interface Certificate {
+  id?: number;
+  supplier: string;
+  certificateType: string;
+  validFrom: string;
+  validTo: string;
+}
+
+const certificateNames: Certificate[] = [
+  {
+    supplier: 'Kellia AG, 1, Berlin',
+    certificateType: 'Permission of Printing',
+    validFrom: '21.08.2017',
+    validTo: '26.08.2017',
+  },
+  {
+    supplier: 'Kamikazi AG, 1, Berlin',
+    certificateType: 'Permission of Printing',
+    validFrom: '21.08.2017',
+    validTo: '26.08.2017',
+  },
+  {
+    supplier: 'Kellia Kamikazi AG, 1, Berlin',
+    certificateType: 'Permission of Printing',
+    validFrom: '21.08.2017',
+    validTo: '26.08.2017',
+  },
+];
+
+const Example1: React.FC = () => {
+  const [certificates, setCertificates] = useState<Certificate[]>([]);
+
+  useEffect(() => {
+    const initializeDB = async () => {
```

Comment:
Really nice that you already connected data to indexedDB. However, this way of handling it is problematic and leads to bugs. On your local instance of app please check: now each time you reload app, default certificates data is added to indexedDB, leading to getting 6 same certificates each time it loads. This is incorrect behavior. This is what you should do:

1. in Certificate interface, you already added id and made it optional. However, each certificate needs to have id to be able to distinguish it from other certificates. So make id required.
2. In your default data, you are already providing all of the data you need, you can as well add id. For example you could use numbers and set them 1,2,3...
3. Inserting of default data should be done only once and only if data is not already in database. To respect data, function that adds certificates to db should check whether this certificate already exists, by id. If it exists, update existing entry. If it doesn't exist, add new entry to database.
4. Initializing default data of app, shouldn't be done in Example1.tsx since this is only component that displays certificates. It has nothing to do with loading default entry set. You could move initialization logic to App.tsx since this is starting point of your application.